api/src/main/java/io/opentelemetry/trace/attributes/AbstractAttribute.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.attributes;
+
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Abstract base class for {@link Attribute} implementations.
+ *
+ * @param <T> the attribute value type
+ */
+@Immutable
+public abstract class AbstractAttribute<T> implements Attribute<T> {
```

Comment:
I suggest naming these `*AttributeSetter` instead of just `*Attribute` to avoid confusion with already existing classes like `io.opentelemetry.trace.AttributeValue`.

Reply:
Changed to AttributeSetter